paddle/fluid/framework/garbage_collector.h

```diff
@@ -118,7 +118,7 @@ void GarbageCollector::Add(Container &&objs, Callback &&callback) {
 
   GarbageQueue *garbage_queue = nullptr;
   {
-    std::lock_guard<std::mutex> guard(mutex_);
+    std::lock_guard<std::mutex> guard(*mutex_);
```

Comment:
Do you mean that if the `max_memory_size_` is little 1, there doesn't need a lock?

Reply:
Exactly.